st2common/st2common/util/misc.py

```diff
@@ -68,7 +85,10 @@ def strip_last_newline_char(input_str):
     if not input_str:
         return input_str
 
-    if input_str.endswith('\n'):
-        return input_str[:-1]
+    if not char_to_strip:
+        return input_str
+
+    if input_str.endswith(char_to_strip):
```

Comment:
rstrip is dangerous because it can remove multiple instances. 

E.g. In [1]: s = 'shit\n\n'

In [2]: s.rstrip('\n')
Out[2]: 'shit'

In [3]: